Extra fragments and clang version

berbascum opened the pull request:
Add EXTRA_FRAGMENTS, CLANG_VERSION and CLANG_CUSTOM support

berbascum committed:
[linux-packaging-snippets] kernel-snippet: add EXTRA_FRAGMENTS support

Multiple config fragments can be supplied at the configs top

Needed changes in kernel-info are included in the
kernel-info.mk.example file

g7 commented on kernel-info.mk.example:
```diff
@@ -198,14 +201,24 @@ BUILD_LLVM = 0
-DEB_TOOLCHAIN = linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, clang-android-6.0-4691093, gcc-4.9-aarch64-linux-android, g++-4.9-aarch64-linux-android, libgcc-4.9-dev-aarch64-linux-android-cross
+DEB_TOOLCHAIN = linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, gcc-4.9-aarch64-linux-android, g++-4.9-aarch64-linux-android, libgcc-4.9-dev-aarch64-linux-android-cross,
+# When using the Droidian clang, add to the build deps
+ifneq ($(CLANG_CUSTOM), 1)
```
I think this can be moved to the actual kernel-snippet right?

berbascum replied:
Yes, that way we would have a cleaner kernel-info.
Are you referring to DEB_TOOLCHAIN as well?

g7 replied:
Heh, DEB_TOOLCHAIN might be a bit more troublesome even if I'd like to hide it as well. But is used to add some other packages on top (some kernel packages for example build-depend on another version of the initramfs).

I would like to see it gone, but I'm okay to keep it around for now... only thing is that when we will add the clang package (if CLANG_CUSTOM is not used), we would probably need to handle cases where the original DEB_TOOLCHAIN misses a comma, as that might happen frequently

berbascum replied:
I agree to keep the DEB_TOOLCHAIN in kernel-info, is useful. Was confused because the above mention. Sorry!

About the CLANG_CUSTOM, you are right.
What do you think about this?

DEB_TOOLCHAIN = linux-initramfs-halium-generic:arm64, binutils-aarch64-linux-gnu, gcc-4.9-aarch64-linux-android, g++-4.9-aarch64-linux-android, libgcc-4.9-dev-aarch64-linux-android-cross

ifneq ($(CLANG_CUSTOM), 1)
DEB_TOOLCHAIN := clang-android-$(CLANG_VERSION), $(DEB_TOOLCHAIN)
endif

berbascum replied:
I pushed to my sid new code with both suggestions: move the CLANG_CUSTOM condition to kernel-snippet and fix the comma problem.

Take a look when you can, thank you!

g7 replied:
thanks a lot, looks great!

berbascum committed:
[linux-packaging-snippets] kernel-info: CLANG_VERSION and CLANG_CUSTOM

CLANG_VERSION simplify the clang version management

CLANG_CUSTOM = 1 excludes clang from DEB_TOOLCHAIN

g7 commented:
Thanks a lot!

berbascum commented:
Thanks to you for accepting!
